prov/gni/test/freelist.c

```diff
@@ -124,6 +124,37 @@ Test(gnix_freelist, freelist_refill_test)
 	_gnix_fl_destroy(&fl);
 }
 
+Test(gnix_freelist, freelist_zero_refill_test)
+{
+	struct gnix_freelist fl;
+	int i, ret;
+	const int num_elems = 71;
+	struct dlist_entry *elems[num_elems + 1];
+	const int refill_size = 0;
+
+	/* non-optimized code may not zero structures */
+	memset(&fl, 0x0, sizeof(struct gnix_freelist));
+
+	ret = _gnix_fl_init(sizeof(struct dlist_entry), 0,
+						num_elems, refill_size, 0, 0, &fl);
+	cr_assert_eq(ret, FI_SUCCESS, "Failed to initialize freelist");
+
+	for (i = 0; i < num_elems; i++) {
+		ret = _gnix_fl_alloc(&elems[i], &fl);
+		cr_assert_eq(ret, FI_SUCCESS, "Failed to obtain dlist_entry");
+	}
+
+	cr_assert(_gnix_fl_empty(&fl), "Freelist not empty");
+
+	ret = _gnix_fl_alloc(&elems[num_elems], &fl);
+	cr_assert_eq(ret, -FI_ECANCELED, "Freelist empty and refill size is zero");
```

Comment:
The statement isn't making sense with your ret code. In this case, I'd expect the freelist to be empty and the refill to be zero if it PASSED, but not if it failed. Remember that the string is printed on failure, not success.